test/Proofs.test.js

```diff
@@ -40,16 +41,32 @@ describe("Proofs", function () {
     })
 
     it("requires proofs with an agreed upon probability", async function () {
+      const samples = 256 // 256 samples avoids bias due to pointer downtime
+
       await proofs.startRequiringProofs(slotId, probability)
+      await advanceTime(period)
       let amount = 0
-      for (let i = 0; i < 100; i++) {
+      for (let i = 0; i < samples; i++) {
         if (await proofs.isProofRequired(slotId)) {
           amount += 1
         }
         await advanceTime(period)
       }
-      let expected = 100 / probability
-      expect(amount).to.be.closeTo(expected, expected / 2)
+
+      const p = 1 / probability // expected probability
+      const alpha = 1 / 1000 // unit test can fail once every 1000 runs
```

Comment:
Please excuse my lack of understanding of proof probabilities. Why are we checking a probability? Why are there allowed failures? In my mind, it seems to make sense that that a node will need to provide a proof every period, that's it. Adding in a probability seems to make it such that there is a chance a node won't need to provide a proof during the period.

Reply:
The problem with testing probabilities involving random sampling is that it is non-deterministic. In this unit test I try to strike a balance between finding errors in the implementation and not having the test fail because of a couple of unlucky throws of the dice.

Reply:
Makes sense, thanks Mark.